src/finalisers/umd.ts

```diff
@@ -117,6 +117,7 @@ export default function umd (
 	}
 
 	const wrapperIntro = `(function (global, factory) {
+			if (typeof global === 'undefined' && typeof window !== 'undefined') global = window;
```

Comment:
It might break down in module scripts and non-Node server environments I guess.

Otherwise `typeof self !== 'undefined' ? self : global` may be a more reliable alternative without being too much longer.

Reply:
> Instead of altering this at the top here, how about altering the trailing wrapper from

You are absolutely right, this makes much more sense. And since we were already doing this anyway I agree that this is actually something we should change.

Just checking in node it seems `self` is not defined there. So I guess your last suggestion

```
}(typeof self !== 'undefined' ? self : global, (function (${args}) {${useStrict}
```

makes the most sense. I did some googling but I did not find any information if there are situations (ancient browsers?) where `window` should be used instead of `self` so I guess this should always work.

Reply:
I've been using this approach for years and it works in most environments, but if companies are running highly custom JS environments then it cacn be an issue if self is defined and global is not.

So it does tradeoff a little bit in reliability (those custom JS engine cases...) for brevity.

The most full-proof method would be:

```js
typeof self !== 'undefined' ?  self : typeof global !== 'undefined' ? global : this
```

Reply:
👍

Reply:
Thanks for the input. Do you want to push to my branch or can I do the change? Where should I place it in the code?

Reply:
It would be great if you want to add the change here, at https://github.com/rollup/rollup/blob/master/src/finalisers/umd.ts#L123.

Note that this should probably be done for the IIFE format as well.

Reply:
Not sure if the IIFE change is correct. Please review and give me feedback.